### PR TITLE
DEVTOOLS: Replace ugly SDL/MinGW hack with a cleaner hack

### PR DESCRIPTION
**Pull request by orgads**
Instead of having `main=SDL_main` defined by `pkg-config`/`sdl2-config --cflags`
and undefining it everywhere, strip it in its origin.

### CODE REVIEW

**Review by lotharsm — APPROVED** [review]
Looking good to me - thanks!

File: configure
Comment (lephilousophe) [inline configure:3793]: Is it really needed here?

File: configure
Comment (lephilousophe) [inline configure:3802]: and here?

**Comment by lephilousophe** [comment]
Overall, I don't understand the idea to request review and not even wait for a day before merging.
I agree PR must not get forgotten but taking some days is not a problem at all.

File: configure
Comment (orgads) [reply configure:3793]: Yes, because SDL_net "requires" sdl2, so it returns its flags too.

File: configure
Comment (orgads) [reply configure:3802]: Not sure about this one, but it shouldn't hurt.

**Comment by orgads** [comment]
Trust me, I'm an engineer. 😉

Just kidding. I'll wait longer next time. One thing I like about gerrit is that a reviewer can vote +1, still requiring another reviewer to vote +2 before merging. A feature I miss on github.

**Pull request merged**
